content/backend/graphql-js/8-filtering-pagination-and-sorting.md

```diff
@@ -76,30 +81,31 @@ query {
 
 ### Pagination
 
-Pagination is a tricky topic in API design. On a high-level, there are two major approaches regarding how it can be tackled:
+Pagination is a tricky topic in API design. On a high-level, there are two major approaches for tackling it:
 
 - **Limit-Offset**: Request a specific _chunk_ of the list by providing the indices of the items to be retrieved (in fact, you're mostly providing the start index (_offset_) as well as a count of items to be retrieved (_limit_)).
 - **Cursor-based**: This pagination model is a bit more advanced. Every element in the list is associated with a unique ID (the _cursor_). Clients paginating through the list then provide the cursor of the starting element as well as a count of items to be retrieved.
 
-Prisma supports both pagination approaches (read more in the [docs](https://www.prisma.io/docs/-rsc2#pagination)). In this tutorial, you're going to implement limit-offset pagination.
+Prisma supports both pagination approaches (read more in the [docs](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/pagination)). In this tutorial, you're going to implement limit-offset pagination.
 
 > **Note**: You can read more about the ideas behind both pagination approaches [here](https://dev-blog.apollodata.com/understanding-pagination-rest-graphql-and-relay-b10f835549e7).
 
-Limit and offset are called differently in the Prisma API:
+Limit and offset have different names in the Prisma API:
 
-- The _limit_ is called `first`, meaning you're grabbing the _first_ x elements after a provided start index. Note that you also have a `last` argument available which correspondingly returns the _last_ x elements.
-- The _start index_ is called `skip`, since you're skipping that many elements in the list before collecting the items to be returned. If `skip` is not provided, it's `0` by default. The pagination then always starts from the beginning of the list (or the end in case you're using `last`).
+// TODO (robin-macpherson): @nikolasburk it seems we still need to update this over in the `grraphql-js` project as well
+- The _limit_ is called `take`, meaning you're "taking" x elements after a provided start index.
```

Comment:
@nikolasburk this was a little weird because the Prisma docs say that this is now called `take` but I was coding along with Journaly side-by-side testing everything out and it seemed to still be `first` and I saw that you still had it as `first` in the completed project code. Is there some discrepancy here?

Reply:
Ah sorry, yes we had a breaking change in the pagination API in [2.0.0-beta.7](https://github.com/prisma/prisma/releases/tag/2.0.0-beta.7):

- Removed `first`, `last`, `before`, `after` arguments.
- Added `cursor` and `take` arguments.
- `skip` argument unchanged.

The `take` argument replaces `first` and `last`.

Where did you see outdated Prisma docs that were using the old pagination API, we should make sure that this gets updated!